proposed/2021/VulnerabilityInfoInPMUI.md

```diff
@@ -0,0 +1,54 @@
+# Expose package vulnerability information in the Package Manager UI - Technical Spec
+
+* Status: **In Review**
+* Author(s): [Jean-Pierre Briedé](https://github.com/jebriede)
+* Issue: [10981](https://github.com/NuGet/Home/issues/10981) Expose package vulnerabilities information in the PMUI
+* Type: Feature
+
+## Problem Background
+
+Developers that install NuGet packages through Visual Studio may be installing packages that have known vulnerabilities or they may have a project that is depending on a vulnerable version of a NuGet package. There is currently no information in the Package Manager UI that indicates that a package version is vulnerable so developers may continue to use a vulnerable package in their project. Users should be able to see a warning about a package version they are browsing or one they have installed so they can be informed of the vulnerability and hopefully update to a different version of the package that is not vulnerable or use a different package.
+
+## Who are the customers
+
+* All users that install a NuGet packages using the Package Manager in Visual Studio.
+* All users that manage their installed NuGet packages using the Package Manager in Visual Studio.
+
+## Requirements
+
+* A warning icon will be shown in the Installed tab and in the package list if the project has any vulnerable packages installed. These icons will be the same ones that are used when deprecated packages are installed. In the case where a package version is both vulnerable and deprecated, only one icon will still be shown, but the tooltip on the icon will reflect the vulnerability and deprecation.
+* Maintain the functionality of the deprecated warning icon in the package list as well as the installed tab. The icon will be used to indicate vulnerabilities, but will need to still indicate deprecated packages.
+* There can be one or more vulnerabilities for a package version so the details pane will need to be able to display a list of vulnerabilities, the severity levels, and the advisory links for each.
+* When multiple package sources have the same package and All package sources is selected, the vulnerability data will need to be merged to ensure vulnerability information is shown for a package if any of the packages sources provides vulnerability information for the given package.
+
+## Goals
+
+* Show the existing warning icon, used for deprecation, in the Installed tab with a tooltip that indicates how many vulnerable packages are installed.
+* Show the existing warning icon, used for deprecation, next to the package version of a vulnerable package in the package list.
+* Show detailed information about the package vulnerabilities in the package details pane when the user clicks on a package.
+* Track via telemetry when an action (installed, update, uninstall) is performed on a package with a vulnerability from within the Package Manager UI.
```

Comment:
In the telemetry PR, you requested changes such that an array of severities is included along with a count of vulnerable packages per action. Can whatever goal prompted that change be made more explicit here?

Reply:
Let's chat offline about this.